db-engine/src/main/java/com/splicemachine/db/catalog/types/IndexDescriptorImpl.java

```diff
@@ -276,9 +274,16 @@ public boolean isUnique()
     /** @see IndexDescriptor#baseColumnPositions */
     public int[] baseColumnPositions()
     {
+        if (isInvalidIndexDescriptor()) {
+            throw new UnsupportedOperationException();
```

Comment:
I would assert here since this should never happen, if you want to keep an exception, then I suggest throwing a more meaningful exception + a message.

Reply:
Done